### ISSUE TITLE
[BUG] `Import-Module -Name PSRule.Rules.Azure` throws warning about `Az.Resources` even with `-WarningAction 'SilentlyContinue'`

### ISSUE THREAD

**Issue by o-l-a-v** [Azure/PSRule.Rules.Azure#2887]
### Description of the issue

Importing `PSRule.Rules.Azure` in an environment without the `Az.Resources` module installed throws a warning `To use PSRule for Azure export cmdlets please install Az.Resources.` even if:

```powershell
PS > Import-Module -Name 'PSRule.Rules.Azure` -WarningAction 'SilentlyContinue'`
WARNING: To use PSRule for Azure export cmdlets please install Az.Resources.
PS >
```

I'd like to hide this warning in my particular scenario.

### Error messages

No error message

### Reproduction

See description

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.36.0
